schema/field/type.go

```diff
@@ -127,6 +127,10 @@ func (t TypeInfo) Comparable() bool {
 	switch t.Type {
 	case TypeBool, TypeTime, TypeUUID, TypeEnum, TypeString:
 		return true
+	case TypeOther:
+		// Assumes the author of the type knows what they are doing and
+		// sets a reasonable schema type for the dialect used.
+		return true
```

Comment:
Maybe we can check if the [`reflect.Type`](https://golang.org/pkg/reflect/#Type) is comparable and store this information on the [schema/field.TypeInfo](https://pkg.go.dev/entgo.io/ent@v0.6.0/schema/field#TypeInfo)? See the `Type.Comparable` doc. 

There are some cases where the Go type is not comparable, but the database type is comparable so maybe in this case we can check if the type implements an interface that we provide?

Can you share what type do you store on `field.Other`?

Reply:
I too thought briefly on using an interface, but creating another Go interface is independent to the output of `Value() (driver.Value, error)` which is what really matters at the dialect/db level.

As for the `field.Other`, I'm usually embedding/wrapping `time.Time`, or custom structs for holding tree-like data, marshaled out to the db as strings, e.g. [ltree](https://www.postgresql.org/docs/current/ltree.html#id-1.11.7.30.8) which when ordered, is returned as a tree traversal.

Reply:
👍 

Thanks and sorry for the delay.